implementation/src/main/java/io/smallrye/mutiny/subscription/DemandPacer.java

```diff
@@ -0,0 +1,88 @@
+package io.smallrye.mutiny.subscription;
+
+import static io.smallrye.mutiny.helpers.ParameterValidation.positive;
+import static io.smallrye.mutiny.helpers.ParameterValidation.validate;
+
+import java.time.Duration;
+
+import io.smallrye.common.annotation.Experimental;
+import io.smallrye.mutiny.Multi;
+
+/**
+ * Interface for {@link Multi} demand pacers and the {@link Multi#paceDemand()} operator.
+ * <p>
+ * A demand-pacer allows controlling upstream demand using a request and a delay.
+ * Each time the delay expires the pacer can evaluate a new demand based on the previous request and the number of emitted items
+ * that have been observed since the previous request.
+ * <p>
+ * The {@link FixedDemandPacer} offers a fixed delay / fixed demand implementation, but custom / adaptive strategies can be
+ * provided through this {@link DemandPacer} interface.
+ */
+@Experimental("Demand pacing is a new experimental API introduced in Mutiny 1.5.0")
+public interface DemandPacer {
+
+    /**
+     * A demand request.
+     */
+    @Experimental("Demand pacing is a new experimental API introduced in Mutiny 1.5.0")
+    class Request {
+
+        private final long demand;
+        private final Duration delay;
+
+        /**
+         * Constructs a new request.
+         *
+         * @param demand the number of items, must be strictly positive
+         * @param delay the delay, must not be {@code null}, must be strictly positive
+         */
+        public Request(long demand, Duration delay) {
+            this.demand = positive(demand, "demand");
+            this.delay = validate(delay, "delay");
```

Comment:
I'm wondering if we should catch FOREVER here.

Reply:
Seems like a good place indeed